packages/@uppy/provider-views/src/utils/getClickedRange.ts

```diff
@@ -0,0 +1,36 @@
+import type {
+  PartialTreeFile,
+  PartialTreeFolderNode,
+} from '@uppy/core/lib/Uppy'
+
+// Shift-clicking selects a single consecutive list of items
+// starting at the previous click.
+const getClickedRange = (
+  clickedId: string,
+  displayedPartialTree: (PartialTreeFolderNode | PartialTreeFile)[],
+  isShiftKeyPressed: boolean,
+  lastCheckbox: string | null,
+): string[] => {
+  const lastCheckboxIndex = displayedPartialTree.findIndex(
+    (item) => item.id === lastCheckbox,
+  )
+
+  if (lastCheckboxIndex !== -1 && isShiftKeyPressed) {
+    const newCheckboxIndex = displayedPartialTree.findIndex(
+      (item) => item.id === clickedId,
+    )
+    const clickedRange = (
+      lastCheckboxIndex < newCheckboxIndex ?
+        displayedPartialTree.slice(lastCheckboxIndex, newCheckboxIndex + 1)
+      : displayedPartialTree.slice(
+          newCheckboxIndex,
+          lastCheckboxIndex + 1,
+        )).map((item) => item.id)
```

Comment:
```suggestion
    const clickedRange = displayedPartialTree.slice(
      Math.min(lastCheckboxIndex, newCheckboxIndex),
      lastCheckboxIndex < newCheckboxIndex ? newCheckboxIndex + 1 : lastCheckboxIndex + 1,
    ).map((item) => item.id)
```

Reply:
Love it 👍

Reply:
I'll incorporate Antoine's suggestion, but in general this is a good example of prettier 80-symbol-line making code outright unreadable, we went from this before prettier:

```js
const clickedRange = (lastCheckboxIndex < newCheckboxIndex ?
    displayedPartialTree.slice(lastCheckboxIndex, newCheckboxIndex + 1)
  : displayedPartialTree.slice(newCheckboxIndex, lastCheckboxIndex + 1)
).map((item) => item.id)
```

to this after prettier:

```js
const clickedRange = (
  lastCheckboxIndex < newCheckboxIndex ?
    displayedPartialTree.slice(lastCheckboxIndex, newCheckboxIndex + 1)
  : displayedPartialTree.slice(
      newCheckboxIndex,
      lastCheckboxIndex + 1,
    )).map((item) => item.id)
```

Our eslint is at 120-symbols limit, I feel we should look into making prettier's config go at `printWidth: 120` too.

Reply:
120 is way too wide imo and uncommon. I'm fine with the current print width.